### PR TITLE
Allow configuration of patroni's replication mode

### PR DESCRIPTION
**Pull request by lalbers**
Related to issue https://github.com/zalando/postgres-operator/issues/505
______
Patroni offers the possibility of a synchronous replication. This ensures consistency across a cluster by confirming that writes are written to a secondary before returning to the connecting client with a success.

It would be great, if this option is adjustable via postgresql.crd

Patroni replication modes:
[zalando/patroni - replication_modes.rst](https://github.com/zalando/patroni/blob/master/docs/replication_modes.rst)

### CODE REVIEW

File: manifests/complete-postgres-manifest.yaml [hunk 1]
```diff
@@ -64,6 +64,7 @@ spec:
     ttl: 30
     loop_wait: &loop_wait 10
     retry_timeout: 10
+    synchronous_mode: true
```
Comment (FxKu) [inline manifests/complete-postgres-manifest.yaml]: better set it to false here

File: docs/reference/cluster_manifest.md [hunk 2]
```diff
@@ -212,6 +212,8 @@ explanation of `ttl` and `loop_wait` parameters.
   automatically created by Patroni for cluster members and permanent replication
   slots. Optional.
 
+* **synchronous_mode**
+  Patroni `synchronous_mode` parameter value. The default is set to `false`. Optional.
```
Comment (FxKu) [inline docs/reference/cluster_manifest.md]:
```suggestion
  Patroni `synchronous_mode` parameter value. Optional.

```

Comment (FxKu) [reply docs/reference/cluster_manifest.md]: There's no default specified in the go code and generally we don't do it for manifest fields. Just say it's optional as we do with other flags e.g. `enableMasterLoadBalancer` (and add a blank line between next header)

File: pkg/cluster/k8sres.go [hunk 3]
```diff
@@ -222,6 +223,9 @@ PatroniInitDBParams:
 	if patroni.Slots != nil {
 		config.Bootstrap.DCS.Slots = patroni.Slots
 	}
+	if patroni.SynchronousMode != false {
```
Comment (FxKu) [inline pkg/cluster/k8sres.go]:
```suggestion
	if patroni.SynchronousMode {
```

As there is no default, this field will be `nil` when left out. So we should only enter this code block when `true`. Right now, I'm not 100% sure if we have to check `patroni.SynchronousMode != nil`, too

File: manifests/complete-postgres-manifest.yaml (hunk 1 above)
Comment (lalbers) [reply manifests/complete-postgres-manifest.yaml]: I agree. Fixed :heavy_check_mark:  https://github.com/zalando/postgres-operator/pull/869/commits/59d567edd0bc0ecf1d17d87566c18935dd21ec1f

File: docs/reference/cluster_manifest.md (hunk 2 above)
Comment (lalbers) [reply docs/reference/cluster_manifest.md]: Does the same but is shorter. So we can customize it if you like.

**Comment by FxKu** [comment]
:+1:

**Comment by FxKu** [comment]
@lalbers as Alex suggested could you add the same for `synchronous_mode_strict`, too?

**Comment by lalbers** [comment]
> @lalbers as Alex suggested could you add the same for `synchronous_mode_strict`, too?

sounds reasonable, i will add that option :rocket:

**Comment by FxKu** [comment]
:+1:

**Comment by sdudoladov** [comment]
:+1:

**Pull request merged**